engine/server/engine/server/engine_connect_server_service.go

```diff
@@ -20,29 +25,37 @@ type EngineConnectServerService struct {
 
 	enclaveManager *enclave_manager.EnclaveManager
 
-	//The protected user ID for metrics analytics purpose
+	// The protected user ID for metrics analytics purpose
 	metricsUserID string
 
-	//User consent to send metrics
+	// User consent to send metrics
 	didUserAcceptSendingMetrics bool
 
-	//The client for consuming container logs from the logs' database server
-	logsDatabaseClient centralized_logs.LogsDatabaseClient
+	// The clients for consuming container logs from the logs' database server
+
+	// per week pulls logs from enclaves created post log retention feature
+	perWeekLogsDatabaseClient centralized_logs.LogsDatabaseClient
+
+	// per file pulls logs from enclaves created pre log retention feature
+	// TODO: remove once users are fully migrated to log retention/new log schema
```

Comment:
If you think that there are no users running long running enclaves (enclaves prior to next release) as of this PR, I can remove this and this PR would be simplified quite a bit

Reply:
If this works already we can keep it, it's good to be fully backward compatible, and we can remove it in a few weeks from now. Maybe stick a TODO somewhere saying that past 2023-09-XX, this can be removed

Reply:
sounds good.